src/dbnode/persist/fs/read.go

```diff
@@ -205,12 +205,12 @@ func (r *reader) Open(opts DataReaderOpenOptions) error {
 	result, err := mmap.Files(os.Open, map[string]mmap.FileDesc{
 		indexFilepath: mmap.FileDesc{
 			File:    &r.indexFd,
-			Bytes:   &r.indexMmap,
+			Bytes:   &r.indexMmap.Bytes,
 			Options: mmap.Options{Read: true, HugeTLB: r.hugePagesOpts},
 		},
 		dataFilepath: mmap.FileDesc{
 			File:    &r.dataFd,
-			Bytes:   &r.dataMmap,
+			Bytes:   &r.dataMmap.Bytes,
```

Comment:
Hm, looks like `dataMmap` and `indexMmap` will never be updated with the `ReporterOptions` so it will never do the correct stat decrease when mmap is closed.

Perhaps we opt for `FileDesc` to take a `*mmap.Descriptor` to populate instead of a `*[]byte` like it does now to avoid this common thing (made a similar suggestion at that callsite too).